Sources/SwiftSyntax/SyntaxVisitor.swift.gyb

```diff
@@ -141,3 +141,75 @@ open class SyntaxVisitor {
     }
   }
 }
+
+public protocol SyntaxTransformVisitor {
+  associatedtype ResultType = Void
```

Comment:
@CodaFi we had discussed having one of these for each kind of thing we might want to transform (decls, exprs, ec.), but I think one is OK for now. If the need arises to have different ones, we can update the visitor.